packages/odoo/test/Adaptor.test.js

```diff
@@ -0,0 +1,89 @@
+import { expect } from 'chai';
+import { enableMockClient } from '@openfn/language-common/util';
+import {
+  execute,
+  create,
+  dataValue,
+  update,
+  read,
+  deleteRecord,
+} from '../src/Adaptor.js';
+
+import MockAgent from './mockAgent.js';
+import { setGlobalDispatcher } from 'undici';
+
+setGlobalDispatcher(MockAgent);
+
+const configuration = {
+  baseUrl: 'https://test.odoo.com/odoo',
+  password: 'somepassword',
+  username: 'someusername',
+  database: 'somedatabase',
+};
+
+const state = { configuration };
+
+describe('read record', () => {
+  it('should read a record', () => {
+    const mock = {
+      read: (model, id, fields) => {
+        expect(model).to.eql('product.product');
+        return {
+          id: 2,
+          name: 'Saas Product',
+        };
+      },
+    };
+    enableMockClient(mock);
+
+    read('product.product', [2], ['name'])(state);
```

Comment:
There is no assertion on what the returned state looks like. How do we know if the adaptor has written to state.data correctly?

Reply:
state only has the config and if I try to log the response, I do not get anything

Reply:
> state only has the config

That sounds like a bug you need to fix :)